src/tools/crawl.ts

```diff
@@ -1,20 +1,27 @@
+// Comment what this file does, and overall add more comment to every file and blocks of code.
+
 import * as fg from 'fast-glob';
 import * as fs from 'fs';
 import * as path from 'path';
 
 import { IGroupDesc } from '../framework/loader.js';
 
+// Could probably be a constant exposed by loader.js
 const specSuffix = '.spec.ts';
 
+// So this just returns the list of all nodes of the suite and their description but without the tree structure?
 export async function crawl(suite: string): Promise<IGroupDesc[]> {
   const specDir = path.normalize(`src/suites/${suite}/`); // Always ends in /
+  // Should this type of thing be asserts?
```

Comment:
Probably, but asserts just throw exceptions and this causes giant ugly UnhandledPromiseRejectionWarning messages.

Reply:
Ack :(